evalml/pipelines/components/transformers/samplers/oversamplers.py

```diff
@@ -0,0 +1,133 @@
+from skopt.space import Integer
+
+from evalml.pipelines.components.transformers.samplers.base_sampler import (
+    BaseSampler
+)
+from evalml.utils import import_or_raise
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+class SMOTESampler(BaseSampler):
+    """SMOTE Oversampler component. Works on numerical datasets only. This component is only run during training and not during predict."""
+    name = "SMOTE Oversampler"
+    hyperparameter_ranges = {
+        'k_neighbors': Integer(2, 10)
```

Comment:
@bchen1116 oops we didn't chat about hyperparameter ranges

For v1, let's not expose any automl hyperparameter ranges for our undersampler and oversampler just yet. I think we need to punt thinking about this over to #1717 .

Reply:
Sounds good!

Reply:
I just filed an issue to track removing hyperparam ranges from the undersampler component: #2104